packages/grafana-data/src/types/transformations.ts

```diff
@@ -28,6 +28,12 @@ export interface DataTransformerInfo<TOptions = any> extends RegistryItemWithOpt
    * @param options
    */
   operator: (options: TOptions, context: DataTransformContext) => MonoTypeOperatorFunction<DataFrame[]>;
+  /**
+   * Function that is present will indicate whether a transformation is applicable 
+   * given the current data.
+   * @param options
+   */
+  applicator?: (data: DataFrame[]) => boolean;
```

Comment:
```suggestion
  isApplicable?: (data: DataFrame[]) => boolean;
```

perhaps too complicated, but we could also consider something like panel suggestions.  where each transformation is passed the data and we get a score+possible config options from it 🤔 

for the minimum version, returning a low score is the same as returning false here

Reply:
I'm very much for this! I'll see what I can do for an MVP 😄

Reply:
Pushed updates which should resolve these